cmds/core/base64/base64.go

```diff
@@ -34,31 +38,39 @@ func do(r io.Reader, w io.Writer, decode bool) error {
 	}
 
 	if _, err := io.Copy(w, r); err != nil {
-		return fmt.Errorf("error %s the data: %v", op, err)
+		return fmt.Errorf("error %s the data: %w", op, err)
 	}
 	return nil
 }
 
-func run(name string, stdin io.Reader, stdout io.Writer, decode bool) error {
-	if name != "-" && len(name) > 0 {
+// run runs the base64 command. Why use ...string?
+// makes testing a tad easier (so we don't have an if in main()
+// allows us, should we wish, in future, to go with using
+// names[1] as out. base64 commands are very nonstandard.
+func run(stdin io.Reader, stdout io.Writer, decode bool, names ...string) error {
+	// fbsd lets you name an output file,
+	switch len(names) {
+	case 0:
+	case 1:
+		name := names[0]
+		if name == "-" {
```

Comment:
nit: base64 in mac also recognies two hyphens 

```
➜  ~ echo hahah | base64
aGFoYWgK
➜  ~ echo hahah | base64 -
aGFoYWgK
➜  ~ echo hahah | base64 --
aGFoYWgK
➜  ~ echo hahah | base64 ---
base64: unrecognized option `---'
Usage:	base64 [-hvDd] [-b num] [-i in_file] [-o out_file]
  -h, --help     display this message
  -Dd, --decode   decodes input
  -b, --break    break encoded string into num character lines
  -i, --input    input file (default: "-" for stdin)
  -o, --output   output file (default: "-" for stdout)
➜  ~
```

Reply:
that's horrible :-)
one is bad enough :-)
I am going to remove - support entirely. If people want a placeholder like that, they can use
/dev/stdin
works on osx and linux ...

Reply:
ok, looks like that is GNU way. GNU binary follows same

```
xuehaohu@worldpeace:~$ echo  hhh | base64
aGhoCg==
xuehaohu@worldpeace:~$ echo  hhh | base64 -
aGhoCg==
xuehaohu@worldpeace:~$ echo  hhh | base64 --
aGhoCg==
xuehaohu@worldpeace:~$ echo  hhh | base64 ---
base64: unrecognized option '---'
Try 'base64 --help' for more information.
```

Reply:
I am going to stand my ground and refuse to continue this -- nonsense, and see how many arrows land in my back.